src/Provider/InvoiceFileProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\InvoicingPlugin\Provider;
+
+use Gaufrette\FilesystemInterface;
+use Sylius\InvoicingPlugin\Entity\InvoiceInterface;
+use Sylius\InvoicingPlugin\Generator\InvoiceFileNameGeneratorInterface;
+use Sylius\InvoicingPlugin\Model\InvoicePdf;
+
+final class InvoiceFileProvider implements InvoiceFileProviderInterface
+{
+    /** @var InvoiceFileNameGeneratorInterface */
+    private $invoiceFileNameGenerator;
+
+    /** @var FilesystemInterface */
+    private $filesystem;
+
+    /** @var string */
+    private $invoicesDirectory;
+
+    public function __construct(
+        InvoiceFileNameGeneratorInterface $invoiceFileNameGenerator,
+        FilesystemInterface $filesystem,
+        string $invoicesDirectory
+    ) {
+        $this->invoiceFileNameGenerator = $invoiceFileNameGenerator;
+        $this->filesystem = $filesystem;
+        $this->invoicesDirectory = $invoicesDirectory;
+    }
+
+    public function provide(InvoiceInterface $invoice): InvoicePdf
+    {
+        $invoiceFileName = $this->invoiceFileNameGenerator->generateForPdf($invoice);
+        $invoiceFile = $this->filesystem->get($invoiceFileName);
```

Comment:
What if there is no invoice saved in the filesystem? Shouldn't it be generated here?

Reply:
And what is more, the current solution is not backwards compatible, as there will be no generated pdfs in the filesystem, so no one will be able to download the previously created invoice after this change.

Reply:
Agreed 👍 It should probably be generated here, to keep it both simple enough and backward compatible